### PR TITLE
Update project code

### PR DESCRIPTION
**Pull request by IgorGanapolsky**
This is a continuation of PR #112

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -53,15 +53,15 @@ android {
...
+    supportLibVersion = '27.0.2'
+    espressoVersion = '3.0.1'
+    androidTestVersion = '1.0.1'
+    hamcrestVersion = '1.3'
+    junitVersion = '4.12'
 }
 
 dependencies {
-    kapt "com.android.databinding:compiler:2.3.2"
+    kapt 'com.android.databinding:compiler:3.0.1'
```
Comment (ataulm) [inline app/build.gradle]: last thing (same comment for the change in top level build.gradle): all the dependency version names are extracted in the `ext` block - maybe you can do the same on this.

re: updating an external PR - does pushing to your branch not work?

**Comment by IgorGanapolsky** [comment]
@ataulm Thanks, I updated it, though I don't see what you mean by **top level build.gradle** (nothing was changed there).

File: build.gradle [hunk 2]
```diff
@@ -17,13 +17,13 @@
...
 buildscript {
-    ext.kotlin_version = '1.1.3-2'
+    ext.kotlin_version = '1.2.21'
     repositories {
         google()
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.0.0-beta2'
+        classpath 'com.android.tools.build:gradle:3.0.1'
```
Comment (ataulm) [inline build.gradle]: by top level, I meant this one, but it was already inlined before your change :man_shrugging:

File: app/src/main/java/com/google/samples/apps/topeka/widget/AvatarView.kt [hunk 3]
```diff
@@ -95,7 +95,7 @@ class AvatarView @JvmOverloads constructor(
         super.onDraw(canvas)
         if (isChecked) {
             ContextCompat.getDrawable(context, R.drawable.selector_avatar).apply {
-                setBounds(0, 0, width, height)
+                this!!.setBounds(0, 0, width, height)
```
Comment (ataulm) [inline app/src/main/java/com/google/samples/apps/topeka/widget/AvatarView.kt]: is this necessary? :thinking: if `setBounds` is an instance method, then the instance can't be null.

Wondering whether the addition of `this` and `!!` throughout the PR is needed.

**Comment by keyboardsurfer** [comment]
With the latest source drop these changes seem not necessary any longer.

**Pull request closed without merging**